### PR TITLE
Use C++

### PR DESCRIPTION
**Pull request by Jupeyy**
Lets finally move to c++, there is no real disadvantage, but there are advantages:
- std::chrono
- no C89 limitations or whatever MSVC uses
- no functions with unspecified parameters

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by def-** [comment]
Fine for me. @Learath2 @heinrich5991 opinions?

**Comment by Learath2** [comment]
I'm fine with it aswell, but should we maybe `typedef int64_t int64` to avoid the massive diff or.do you think we should get rid of it sooner than later?

**Comment by Jupeyy** [comment]
Guess as sooner the better it is. Also prevents possible collision on haiku
It's also ez to rebase against it, just replace matching string

**Comment by heinrich5991** [comment]
LGTM.

Can you keep `extern "C" {` for system.h/c? AFAIK this only affects symbol names, and we kind of need that because the antibot uses those symbols.

**Comment by Jupeyy** [comment]
I wonder if we can debug, why CodeQL fails so often.
It says make fails, but dunno what it injects

**Comment by def-** [comment]
If not, then let's just disable it for now.

File: src/base/unicode/tolower.cpp [hunk 1]
```diff
@@ -9,14 +9,16 @@ static int compul(const void *a, const void *b)
...
 	struct UPPER_LOWER *res;
 	key.upper = code;
-	res = bsearch(&key, tolower, NUM_TOLOWER, sizeof(struct UPPER_LOWER), compul);
+	res = (UPPER_LOWER *)bsearch(&key, tolower, NUM_TOLOWER, sizeof(struct UPPER_LOWER), compul);
 
 	if(res == NULL)
 		return code;
 	return res->lower;
 }
+}
```
Comment (heinrich5991) [inline src/base/unicode/tolower.cpp:24]: Missing newline at end of file.

**Comment by def-** [comment]
Let's get this in?

**Comment by Learath2** [comment]
LGTM, looks like we all agree on this

bors r+

**Comment by Learath2** [comment]
Eh, I hoped the conflict would be fine if merged on another branch. @Jupeyy fix the conflict so we can get this in

**Pull request merged**